versioning/BRM/extentmap.cpp

```diff
@@ -5109,23 +4848,27 @@ void ExtentMap::getExtents(int OID, vector<struct EMEntry>& entries,
 
     grabEMEntryTable(READ);
     grabEMIndex(READ);
-    emEntries = fEMShminfo->allocdSize / sizeof(struct EMEntry);
-    // Pre-expand entries to stop lots of small allocs
-    entries.reserve(emEntries);
+    entries.reserve(fExtentMapRBTree->size());
+
     if (incOutOfService)
     {
-        for (i = 0 ; i < emEntries; i++)
-            if ((fExtentMap[i].fileID == OID) &&
-                    (fExtentMap[i].range.size != 0))
-                entries.push_back(fExtentMap[i]);
+        for (auto emIt = fExtentMapRBTree->begin(), end = fExtentMapRBTree->end(); emIt != end;
```

Comment:
This method is an interesting one so let's discuss how to leverage index here.
The most straightforward way is to find all lbids that belongs to the target OID iterating over all dbroots available. And then iterate over lbids running findByLBID() in a loop. However this approach will be slow given a big table with lots of extents. It would be better to find the topmost tree iterator and start iterating from there but I am not sure how to find a suitable first iterator to be honest.